File: client/src/app/services/atm.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs/index";

@Injectable({
  providedIn: 'root'
})
export class AtmService {

  private baseUrl = 'api/atm';

  constructor(private http: HttpClient) {
  }

  getBalance() {

    const url = this.baseUrl + "/1/balance";
    return this.http.get<number>(url,{responseType: 'number' as 'json'});
  }

  deposit(amount:number) {

    const url = this.baseUrl + "/1/deposit?amount="+amount;
    return this.http.put(url,{});
  }

  withdraw(amount:number) {

    const url = this.baseUrl + "/1/withdraw?amount="+amount;
    return this.http.put(url,{},{responseType: 'text' as 'json'});
  }
}

File: src/main/java/com/atm/AtmService.java
package com.atm;


import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class AtmService {

    @Autowired
    AtmRepository atmRepository;


    public Integer getBalance(Integer id){

        return atmRepository.getbalance(id);
    }

    public void deposit(Integer id,Integer amount){

        atmRepository.deposit(id,amount);
    }

    public String withdraw(Integer id,Integer amount){

       return atmRepository.withdraw(id,amount);

    }
}
